src/definitions/validity_info.rs

```diff
@@ -101,20 +106,27 @@ impl TryFrom<ValidityInfo> for CborValue {
             insert_date!(map, expected_update, "expectedUpdate");
         }
 
-        Ok(CborValue::Map(map))
+        Ok(ciborium::Value::Map(map))
     }
 }
 
-impl TryFrom<CborValue> for ValidityInfo {
+impl TryFrom<ciborium::Value> for ValidityInfo {
     type Error = Error;
 
-    fn try_from(v: CborValue) -> Result<ValidityInfo> {
-        if let CborValue::Map(mut map) = v {
+    fn try_from(v: ciborium::Value) -> Result<ValidityInfo> {
+        if let ciborium::Value::Map(map) = v.clone() {
+            let mut map: BTreeMap<String, ciborium::Value> = map
+                .into_iter()
+                .map(|(k, v)| {
+                    let k = k.into_text().map_err(|_| Error::CborError)?;
+                    Ok((k, v))
+                })
+                .collect::<Result<BTreeMap<_, _>>>()?;
```

Comment:
There's quite a few places in this PR where you are converting a vec into a map to index into it. This is a computationally inefficient and could be handled by treating the vec as an iterator and filtering. You don't need to address this as part of this PR, but this is something you could consider coming back to.

Reply:
thank you, I noted and will come back with another PR when time permits